tests/controller/controller_test.go

```diff
@@ -0,0 +1,628 @@
+package controller_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/awslabs/aws-s3-csi-driver/pkg/podmounter/mppod"
+)
+
+var _ = Describe("Mountpoint Controller", func() {
+	Context("Static Provisioning", func() {
+		Context("Scheduled Pod with pre-bound PV and PVC", func() {
```

Comment:
Are we effectively testing that if there's a volume with the S3 CSI Driver, and it is bound to a pod that is scheduled, `waitAndVerifyMountpointPodFor`, otherwise, `expectNoMountpointPodFor`?

Reply:
If so, then a property based test using a state machine might be useful here to ensure all combinations are reached

Reply:
Yes, those are the expectations we assert during the tests. I think we can look into using property tests, but I think I'd also like to keep these known/expected cases here